### PR TITLE
Feature/conflicts in promote

### PR DESCRIPTION
**Pull request by lariciamota**
#### What is the purpose of this pull request?
Add conflict handling to `vtex workspace promote`. Also, some log/success messages were updated to fit the designer's prototype, you can check it out [here](https://www.figma.com/file/SR9a9fT1W6dh2EUri6RPGL/Commands-Flow?node-id=12%3A106) and about the colors [here](https://www.figma.com/file/GpNy0htTpJ7uvw7WKqGdLH/Toolbelt-DX-review?node-id=93%3A21).

#### What problem is this solving?
Changes in workspace are being lost when `vtex workspace promote` is run if conflicts are detected, as conflicts are fixed only when the workspace is opened in the browser. Therefore, if the user run promote without previously opening the workspace in browser their changes are replaced with the masters (_master wins_ strategy), but we want to use the _mine wins_ strategy -> this is the conflict handling implemented in this PR.

You can see more on this [task](https://vtex-dev.atlassian.net/browse/DT-18?atlOrigin=eyJpIjoiNGViMTE4NDg4ZjhlNDU5OWJlN2ZmY2JmNTExOTgyZTYiLCJwIjoiaiJ9).

#### How should this be manually tested?
Run the project in this branch, then after running `yarn watch` follow the steps:

1. Create a new production workspace: `vtex-test workspace create your-workspace-here --production`
2. Use this new workspace: `vtex-test workspace use your-workspace-here`
3. Make some change using your workspace https://workspace--storecomponents.myvtex.com/admin/cms/site-editor
4. Make some change in the same component now using master https://storecomponents.myvtex.com/admin/cms/site-editor (Remember not to open your workspace in the browser)
5. (Optional) If you want to test the request to see if conflict is detected: `http://infra.io.vtex.com/vbase/v0/storecomponents/your-workspace-here/buckets/vtex.pages-graphql/userData/files/store/content.json` using this header: `X-Vtex-Detect-Conflicts: true`. It should return 409.
6. Test the `vtex-test workspace promote` (run twice to answer no and then yes to the confirmation prompt to see both flows)
7. Check https://storecomponents.myvtex.com/ to see if the changes you make on your workspace are now on master.

#### Screenshots or example usage
- "No" flow:
Screen Shot 2020-08-04 at 13 57 52

- "Yes" flow:
Screen Shot 2020-08-04 at 13 58 34

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [x] Update `CHANGELOG.md`

### CODE REVIEW

File: src/modules/workspace/promote.ts [hunk 1]
```diff
@@ -1,45 +1,108 @@
...
         'production'
-      )} workspace\nOnly production workspaces may be promoted\nUse the command ${chalk.blue(
+      )} workspace\nOnly production workspaces may be promoted\nUse the command ${ColorifyConstants.COMMAND_OR_VTEX_REF(
         'vtex workspace create <workspace> --production'
       )} to create a production workspace`
     )
   }
+
+  const spinner = ora('Preparing the workspace to be promoted').start()
+  spinner.color = 'magenta'
```
Comment (Jeymisson) [inline src/modules/workspace/promote.ts:49]: Could this color also be a constant?

File: src/modules/workspace/promote.ts [hunk 2]
```diff
@@ -1,45 +1,108 @@
...
+  await handleConflict()
+  spinner.succeed()
 }
+
 const promptPromoteConfirm = (workspace: string): Promise<boolean> =>
-  promptConfirm(`Are you sure you want to promote workspace ${chalk.green(workspace)} to master?`, true)
+  promptConfirm(
+    `Do you want to promote ${ColorifyConstants.ID(workspace)} to master? ${chalk.dim(
```
Comment (Jeymisson) [inline src/modules/workspace/promote.ts:56]: Should we start grouping those string messages on a messages / strings constants file?

File: src/modules/workspace/promote.ts (hunk 1 above)
Comment (lariciamota) [reply src/modules/workspace/promote.ts:49]: Sure! ora.Color only accepts some specific strings, not hex code like I was doing, but I'll add `magenta` in the constants file 😄

File: src/modules/workspace/promote.ts (hunk 2 above)
Comment (lariciamota) [reply src/modules/workspace/promote.ts:56]: Yes, I think we should! I'll start this in this PR then

File: src/modules/workspace/promote.ts [hunk 3]
```diff
@@ -1,45 +1,74 @@
...
 const promptPromoteConfirm = (workspace: string): Promise<boolean> =>
-  promptConfirm(`Are you sure you want to promote workspace ${chalk.green(workspace)} to master?`, true)
+  promptConfirm(Messages.PROMOTE_PROMPT_CONFIRM(workspace), true)
 
 export default async () => {
-  log.debug('Promoting workspace', currentWorkspace)
+  log.debug(Messages.PROMOTE_INIT, currentWorkspace)
+
   await isPromotable(currentWorkspace)
+  console.log(Messages.PROMOTE_CHECK_WORKSPACE(currentWorkspace, workspaceUrl))
```
Comment (Jeymisson) [inline src/modules/workspace/promote.ts:56]: Why some log.info and some console.log?

File: src/lib/constants/Messages.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    )} If you are looking for other workspaces, run ${ColorifyConstants.COMMAND_OR_VTEX_REF('vtex workspace list')}.\n`,
+  PROMOTE_SUCCESS: (workspace: string) =>
+    `✨ ${chalk.bold('Success!')} The workspace ${ColorifyConstants.ID(
+      workspace
+    )} was promoted to ${ColorifyConstants.ID(
+      'master'
+    )}, taking your changes to the final users. All the content it had is now at the workspace ${ColorifyConstants.ID(
+      'master'
+    )} and the workspace ${ColorifyConstants.ID(workspace)} was deleted.`,
+  PROMOTE_ASK_FEEDBACK: `What do you think about checking the workspace before promote? Please, tell us ${formatHyperlink(
```
Comment (Jeymisson) [inline src/lib/constants/Messages.ts:28]: I think `What do you think about checking the workspace before promote` => `What do you think about checking the workspace before promoting` not sure though.

File: src/modules/workspace/promote.ts (hunk 3 above)
Comment (Jeymisson) [reply src/modules/workspace/promote.ts:56]: Might get it, you don't want the info: tag.

Comment (lariciamota) [reply src/modules/workspace/promote.ts:56]: Yes, also we don't need this info in the debug file

**Pull request merged**